### PR TITLE
[CARBONDATA-1618] Fix issue of not support table comment

### PR DESCRIPTION
**Pull request by chenerlu**
Background: Current carbon do not support table comment when create table.
This PR will support table comment.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/681/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1318/

**Comment by chenerlu** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/682/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1319/

**Comment by chenerlu** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/689/

**Comment by chenliang613** [comment]
@sounakr  please review it.

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1324/

File: integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala [hunk 1]
```diff
@@ -247,7 +247,8 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
       , tableName: String, fields: Seq[Field],
       partitionCols: Seq[PartitionerField],
       tableProperties: mutable.Map[String, String],
-      bucketFields: Option[BucketFields], isAlterFlow: Boolean = false): TableModel = {
+      bucketFields: Option[BucketFields], isAlterFlow: Boolean = false,
+      comment: Option[String] = None): TableModel = {
```
[the remark this answers is not in the document]
Comment (chenerlu) [reply integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala]: Carbon already support column comment.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateTableWithTableComment.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    sql(
+      s"""
+         | create table withoutTableComment(
+         | id int,
+         | name string
+         | )
+         | STORED BY 'carbondata'
+       """.stripMargin
+    )
+
```
[the remark this answers is not in the document]
Comment (chenerlu) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateTableWithTableComment.scala]: This PR not contains this functions.

File: integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (chenerlu) [reply integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala]: Have renamed.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/701/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1337/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateTableWithTableComment.scala [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    val result = sql("describe formatted withoutTableComment")
+
+    checkExistence(result, true, "Comment:")
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateTableWithTableComment.scala]: Can you assert the string include "This table has table comment" also?

Comment (chenerlu) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateTableWithTableComment.scala]: Done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/752/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1389/

**Comment by ravipesala** [comment]
@chenerlu how the tablecomment added through ddl is persisted? I don't see anywhere the logic to persist the tablecommnet.

**Comment by chenerlu** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/789/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1422/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/790/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1423/

File: core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java [hunk 4]
```diff
@@ -79,6 +79,9 @@
   // this idenifier is a lazy field which will be created when it is used first time
   private AbsoluteTableIdentifier identifier;
 
+  // table comment
+  private String tableComment;
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java]: no need to add this attribute as it is already stored in properties

**Comment by ravipesala** [comment]
LGTM, comments of this PR will be handled in another PR

File: core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java (hunk 4 above)
Comment (pawanmalwal) [reply core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java]: Handled it in  https://github.com/apache/carbondata/pull/1472

**Pull request closed without merging**